### PR TITLE
Topic/nginx base

### PR DESCRIPTION
**Pull request by ohdearaugustin**
Add nginx_base role to collection.

### CODE REVIEW

**Comment by ohdearaugustin** [comment]
Blocked by #52

File: roles/apt_repo_ubuntu_nginx/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    - name: Install docker repo key on host
+      ansible.builtin.copy:
+        # Source: https://keyserver.ubuntu.com/pks/lookup?fingerprint=on&op=index&search=0x14AA40EC0831756756D7F66C4F4EA0AAE5267A6C
+        # converted using `gpg --dearmor nginx.gpg`
+        # Unfortunately, this key is only 1024 bits rsa
+        src: nginx.gpg
+        dest: /etc/apt/trusted.gpg.d/nginx.gpg
+      notify: Update apt cache
+
+    - name: Add docker apt repo
```
Comment (MarkusTeufelberger) [inline roles/apt_repo_ubuntu_nginx/tasks/main.yml:23]: docker --> nginx

File: roles/nginx_base/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  notify: Restart nginx
+
+- name: Remove nginx default site
+  ansible.builtin.file:
+    name: /etc/nginx/sites-enabled/default
+    state: absent
+  notify: Restart nginx
+
+- name: Install nginx config.d files
+  when: nginx_base_conf_d_files is defined
```
Comment (MarkusTeufelberger) [inline roles/nginx_base/tasks/main.yml:71]: this is a default var

Comment (MarkusTeufelberger) [reply roles/nginx_base/tasks/main.yml:71]: len > 0?

File: roles/nginx_base/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  loop: "{{ nginx_base_conf_d_files }}"
+  ansible.builtin.copy:
+    src: conf.d/{{ item }}.conf
+    dest: /etc/nginx/conf.d/
+  notify: Restart nginx
+
+- name: Install nginx config snippets
+  when: nginx_base_snippets is defined
```
Comment (MarkusTeufelberger) [inline roles/nginx_base/tasks/main.yml:79]: see above

**Review by MarkusTeufelberger — CHANGES_REQUESTED** [review]
reviewed nginx_base

File: roles/nginx_base/molecule/nginx-extras/verify.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      - snippets/cf-origin-pull
+      - snippets/tls
+      - snippets/tls-legacy
+      - snippets/hsts
+      - snippets/security-headers
+      - snippets/proxy-nobuff
+      - snippets/proxy-forward-headers
+      - snippets/acmetool
+  tasks:
+    - name: Check if nginx is installed
```
[the remark this answers is not in the document]
Comment (ohdearaugustin) [reply roles/nginx_base/molecule/nginx-extras/verify.yml:19]: next time

File: roles/nginx_base/files/snippets/tls.conf [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# ssl security config for tls1.3 by mgit
+# 20200710 - intermediate security nginx config https://mozilla.github.io/server-side-tls/ssl-config-generator/
+ssl_session_timeout  1d;
+ssl_session_cache shared:SSL:10m;
+ssl_session_tickets off;
+
+ssl_dhparam /etc/ssl/dhparams.pem;
+
+ssl_protocols TLSv1.2 TLSv1.3;
+ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384;
```
[the remark this answers is not in the document]
Comment (ohdearaugustin) [reply roles/nginx_base/files/snippets/tls.conf:10]: Regenerated with nginx and openssl version from Debian 10 as this is the oldest OS we use.

**Pull request merged**